libraries/RC_Channel/RC_Channels.cpp

```diff
@@ -70,6 +70,7 @@ bool RC_Channels::read_input(void)
 {
     if (hal.rcin->new_input()) {
```

Comment:
grab, miilis before hand to save getting it twice. 

eg:
`const uint32_t now = AP_HAL::millis()`

Reply:
ok, I think I got everything.  
- I put the new method `last_rc_receiver_input_ms` next to` last_input_ms`, and added comments to describe both.
- I removed the initialization to zero.
- I added your suggestion to avoid grabbing millis() twice.